### ISSUE TITLE
No convergence - Training AdderNet with Cifar10 

### ISSUE THREAD

**Issue by shl-shawn** [AI-Huang/AdderNet-tf#1]
Hi Kan,

Thank you for your brilliant work for writing AdderNet in Tensorflow. 

I learned from [your codes](https://github.com/AI-Huang/AdderNet-tf/blob/becbfe5299b6804f36bca3a95986ce65ae74f62c/models/addernet_cifar10.py#L25) that you have trained ResNet20-V1 and achieved 92.16% accuracy in Cifar10. 

I tried to train the model with the **following command**, but found that it did not converge at all. The training loss and accuracy at 100 epoch are same as that of the first 5 epoch, whose screenshots are attached below.

`python train_addernet_cifar10.py --data_preprocessing "subtract_pixel_mean" --lr_schedule "cifar10_scheduler"  --dataset "cifar10" --use_addernet --data_preprocessing subtract_pixel_mean  --epochs 300 --batch_size 64`

I would be appreciate it if you could let me know what changes I should make.

Best Regard,
Shawn

Epoch 1-5:
![截屏2022-12-15 00 32 48](https://user-images.githubusercontent.com/120353630/207780530-3551ce2b-20bb-48df-9998-12cb7942903a.png)

Epoch 100:

![截屏2022-12-15 00 36 44](https://user-images.githubusercontent.com/120353630/207781063-9bd75590-afd1-45e4-9350-e631d2959a1b.png)

**Comment by AI-Huang** [comment]
Hi Shawn,

Thanks for your trial on my codes. Actually, pls check the [README.md](https://github.com/AI-Huang/AdderNet-tf/blob/main/README.md), which states that this code repository has NOT been tested yet.

This also means your output results are the same as mine, which I haven't uploaded and I am still fixing on.

The reasons I guess for the model not getting convergence could be:

1. TensorFlow's SGDW optimizer is not the same as PyTorch's SGD with momentum;
2. My implementation of the Adder2D layer especially its **gradient function** has some bugs.

I'll try to figure out the problem when I have some time, thanks!
